spec/lib/mobile_application_platform/security_token/service_spec.rb

```diff
@@ -12,23 +12,73 @@
     let(:log_prefix) { '[MobileApplicationPlatform][SecurityToken][Service]' }
 
     shared_examples 'STS token request' do
-      context 'when an issue occurs with the client request' do
-        let(:expected_error) { Common::Client::Errors::ClientError }
-        let(:expected_error_message) do
-          "#{log_prefix} token failed, client error, status: #{status}, description: #{description}," \
-            " application: #{application}, icn: #{icn}"
+      shared_examples 'Client Errors' do
+        context 'when an issue occurs with the client request' do
+          let(:expected_error) { Common::Client::Errors::ClientError }
+          let(:expected_error_message) do
+            "#{log_prefix} Token failed, client error, status: #{status}," \
+              " application: #{application}, icn: #{icn}, context: #{expected_error_body}"
+          end
+
+          before do
+            allow_any_instance_of(described_class).to receive(:perform).and_raise(raised_error)
```

Comment:
I know the previous specs just raised errors to simulate failed responses, but now that we're doing this 'properly' we should have these responses in VCR and replay them in these specs. Refer to `spec/lib/mobile_application_platform/sign_up/service_spec.rb` for an example of what this looks like